src/utility/serializable.hpp

```
@@ -0,0 +1,176 @@
+} // namespace details
+
+struct Serializable {
+    template <typename Metas, std::size_t... Idx>
+    constexpr auto make_serializable_impl(Metas metas, std::index_sequence<Idx...>) const {
+        return details::Serializable {
+            details::MemberMeta { std::get<Idx * 2>(metas), std::get<Idx * 2 + 1>(metas) }...,
+        };
+    }
+    template <typename Metas>
+    constexpr auto make_serializable(Metas metas) const {
+        constexpr auto N = std::tuple_size_v<Metas>;
```

Clang (14.0.6)

[warning] invalid case style for variable 'N'

(readability-identifier-naming)